DEVELOPING_OPAL/demos/src/main/scala/org/opalj/ai/ExceptionUsage.scala

```diff
@@ -78,64 +76,53 @@ object ExceptionUsage extends ProjectAnalysisApplication {
-                    case result.domain.MultipleReferenceValues(singleOriginReferenceValues) =>
-                        singleOriginReferenceValues.foreach { v =>
-                            updateUsageKindForValue(v, usageKind)
-                        }
-                    case _ => /*not relevant*/
-                }
+                usageKind: UsageKind.Value): Unit = value match {
+                case v: result.domain.SingleOriginReferenceValue => updateUsageKindForValue(v, usageKind)
+                case result.domain.MultipleReferenceValues(singleOriginReferenceValues) =>
+                    singleOriginReferenceValues.foreach { v => updateUsageKindForValue(v, usageKind) }
+                case _ => /*not relevant*/
             }
 
             body.iterate { (pc, instruction) =>
                 val operands = result.operandsArray(pc)
                 if (operands != null) { // the instruction is reached...
                     instruction match {
-                        case ATHROW =>
-                            updateUsageKind(operands.head, UsageKind.IsThrown)
+                        case ATHROW => updateUsageKind(operands.head, UsageKind.IsThrown)
                         case i: MethodInvocationInstruction =>
                             val methodDescriptor = i.methodDescriptor
-                            val parametersCount = methodDescriptor.parametersCount
+                            val parametersCount  = methodDescriptor.parametersCount
                             operands.take(parametersCount).foreach(updateUsageKind(_, UsageKind.UsedAsParameter))
                             if (i.isVirtualMethodCall) {
                                 updateUsageKind(operands.drop(parametersCount).head, UsageKind.UsedAsReceiver)
                             }
-                        case _: FieldWriteAccess =>
-                            updateUsageKind(operands.head, UsageKind.StoredInField)
-                        case ARETURN =>
-                            updateUsageKind(operands.head, UsageKind.IsReturned)
-                        case AASTORE =>
-                            updateUsageKind(operands.head, UsageKind.StoredInArray)
-                        case _ =>
+                        case _: FieldWriteAccess => updateUsageKind(operands.head, UsageKind.StoredInField)
+                        case ARETURN             => updateUsageKind(operands.head, UsageKind.IsReturned)
+                        case AASTORE             => updateUsageKind(operands.head, UsageKind.StoredInArray)
+                        case _                   =>
                         /*nothing to do*/
                     }
                 }
             }
 
             val usages =
-                for { ((pc, typeName), exceptionUsage) <- exceptionUsages }
-                    yield ExceptionUsage(method, pc, typeName, exceptionUsage)
+                for { ((pc, typeName), exceptionUsage) <- exceptionUsages } yield ExceptionUsage(method,
```

Comment:
This doesn't look nice, keeping the constructor/method call together would be good if possible.